src/main/java/com/tricentis/neoload/EventsCollector.java

```diff
@@ -101,27 +99,20 @@ private String getRequestStatusLine(SampleResult sampleResult) {
      * - ELEMENT => verify if !TransactionController.isFromTransactionController(result) and set result.getSampleLabel()
      * - Assertion results => View proposed implementation below.
      */
-    private EventDetails getErrorDetails(final SampleResult result) {
+    private ErrorDetails getErrorDetails(final SampleResult result) {
         final String userId = getThreadGroupName(result); // Example: Thread Group A 1-25 => Thread group name + ' ' + thread group instance + "-" + thread (vu) number
         final int instanceId = 1; // We have no a better value => It's a number
 
-        final ErrorDetailsBuilder detailsBuilder = ErrorDetailsBuilder.builder();
-        detailsBuilder
-                .addAttribute(USER_PATH_ID, userId)
-                .addAttribute(USER_PATH_INSTANCE, ValueNumber.of(instanceId))
-                .addAttribute(REQUEST_DURATION, ValueNumber.of(result.getTime()));
-
-        addStringValueIfNotNull(detailsBuilder, ErrorDetailAttribute.REQUEST_STATUS_LINE, getRequestStatusLine(result));
-        addStringValueIfNotNull(detailsBuilder, ErrorDetailAttribute.REQUEST_HEADERS, result.getRequestHeaders());
-        addStringValueIfNotNull(detailsBuilder, ErrorDetailAttribute.RESPONSE_HEADERS, result.getResponseHeaders());
-
-        if (result.getParent() != null) {
-            detailsBuilder.addAttribute(TRANSACTION, result.getParent().getSampleLabel());
-        }
-        detailsBuilder.addAttribute(REQUEST, result.getSampleLabel());
-        detailsBuilder.assertionResults(errorEntryToAssertionResults(result));
-
-        return detailsBuilder.build();
+        return ImmutableErrorDetails.builder()
+            .userPathId(userId)
+            .userPathInstance(ValueNumber.of(instanceId))
+            .requestDuration(ValueNumber.of(result.getTime()))
+            .requestStatusLine(Optional.ofNullable(getRequestStatusLine(result)))
+            .requestHeaders(Optional.of(result.getRequestHeaders()))
+            .responseHeaders(Optional.of(result.getResponseHeaders()))
```

Comment:
Based on previous impl, feels like those fields could be nullable
```suggestion
            .requestHeaders(Optional.ofNullable(result.getRequestHeaders()))
            .responseHeaders(Optional.ofNullable(result.getResponseHeaders()))
```

Reply:
done